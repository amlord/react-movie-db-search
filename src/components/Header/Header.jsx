import React from 'react';
import styled from 'styled-components';
import Wrapper from 'components/Wrapper';

const StyledHeader = styled.header`
  background-color: #081c24;
  color: #fff;
`;

const StyledH1 = styled.h1`
  margin: 7px 0 0;
`;

const StyledImg = styled.img`
  height: 72px;
  vertical-align: top;
`;

const StyledTitleContainer = styled.div`
  height: 72px;
  display: inline-block;
  vertical-align: top;
  padding-left: 20px;
`;

const Header = () => (
  <StyledHeader>
    <Wrapper>
      <StyledImg src="https://www.themoviedb.org/assets/1/v4/logos/powered-by-square-green-11c0c7f8e03c4f44aa54d5e91d9531aa9860a9161c49f5fa741b730c5b21a1f2.svg" alt="The Movie DB logo" />
      <StyledTitleContainer>
        <StyledH1>The Movie DB</StyledH1>
        <small>Discover Movies, TV Shows &amp; People</small>
      </StyledTitleContainer>
    </Wrapper>
  </StyledHeader>
);

export default Header;
