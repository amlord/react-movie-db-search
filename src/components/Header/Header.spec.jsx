import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

describe('Header', () => {
  it('displays correctly', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });
});
