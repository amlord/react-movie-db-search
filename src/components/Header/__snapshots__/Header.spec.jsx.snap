// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header displays correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Wrapper>
        <Header__StyledImg
          alt="The Movie DB logo"
          src="https://www.themoviedb.org/assets/1/v4/logos/powered-by-square-green-11c0c7f8e03c4f44aa54d5e91d9531aa9860a9161c49f5fa741b730c5b21a1f2.svg"
        />
        <Header__StyledTitleContainer>
          <Header__StyledH1>
            The Movie DB
          </Header__StyledH1>
          <small>
            Discover Movies, TV Shows & People
          </small>
        </Header__StyledTitleContainer>
      </Wrapper>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Header__StyledImg
            alt="The Movie DB logo"
            src="https://www.themoviedb.org/assets/1/v4/logos/powered-by-square-green-11c0c7f8e03c4f44aa54d5e91d9531aa9860a9161c49f5fa741b730c5b21a1f2.svg"
          />,
          <Header__StyledTitleContainer>
            <Header__StyledH1>
              The Movie DB
            </Header__StyledH1>
            <small>
              Discover Movies, TV Shows & People
            </small>
          </Header__StyledTitleContainer>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alt": "The Movie DB logo",
            "src": "https://www.themoviedb.org/assets/1/v4/logos/powered-by-square-green-11c0c7f8e03c4f44aa54d5e91d9531aa9860a9161c49f5fa741b730c5b21a1f2.svg",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Header__StyledH1>
                The Movie DB
              </Header__StyledH1>,
              <small>
                Discover Movies, TV Shows & People
              </small>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "The Movie DB",
              },
              "ref": null,
              "rendered": "The Movie DB",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Discover Movies, TV Shows & People",
              },
              "ref": null,
              "rendered": "Discover Movies, TV Shows & People",
              "type": "small",
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Wrapper>
          <Header__StyledImg
            alt="The Movie DB logo"
            src="https://www.themoviedb.org/assets/1/v4/logos/powered-by-square-green-11c0c7f8e03c4f44aa54d5e91d9531aa9860a9161c49f5fa741b730c5b21a1f2.svg"
          />
          <Header__StyledTitleContainer>
            <Header__StyledH1>
              The Movie DB
            </Header__StyledH1>
            <small>
              Discover Movies, TV Shows & People
            </small>
          </Header__StyledTitleContainer>
        </Wrapper>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Header__StyledImg
              alt="The Movie DB logo"
              src="https://www.themoviedb.org/assets/1/v4/logos/powered-by-square-green-11c0c7f8e03c4f44aa54d5e91d9531aa9860a9161c49f5fa741b730c5b21a1f2.svg"
            />,
            <Header__StyledTitleContainer>
              <Header__StyledH1>
                The Movie DB
              </Header__StyledH1>
              <small>
                Discover Movies, TV Shows & People
              </small>
            </Header__StyledTitleContainer>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alt": "The Movie DB logo",
              "src": "https://www.themoviedb.org/assets/1/v4/logos/powered-by-square-green-11c0c7f8e03c4f44aa54d5e91d9531aa9860a9161c49f5fa741b730c5b21a1f2.svg",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Header__StyledH1>
                  The Movie DB
                </Header__StyledH1>,
                <small>
                  Discover Movies, TV Shows & People
                </small>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "The Movie DB",
                },
                "ref": null,
                "rendered": "The Movie DB",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Discover Movies, TV Shows & People",
                },
                "ref": null,
                "rendered": "Discover Movies, TV Shows & People",
                "type": "small",
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
